kolibri/core/content/management/commands/deletecontent.py

```diff
@@ -127,9 +127,20 @@ def handle_async(self, *args, **options):
         delete_all_metadata = delete_metadata(
             channel, node_ids, exclude_node_ids, force_delete
         )
-
+        nodes = ContentNode.objects.filter(id__in=node_ids).all()
+        lst_nodes = list(nodes)
         unused_files = LocalFile.objects.get_unused_files()
-
+        total_resource_number, unused_files_, total_bytes_to_transfer = get_import_export_data(
```

Comment:
In some cases, the `delete_all_metadata` command is already invoking this function - it might be simpler to change what is happening inside that function to always invoke this, and then to return both the `delete_all_metadata` flag, and the `total_resource_number`.

One consideration for this is that this function takes a very long time, so if we can avoid calling it more than we need to, that is preferable.

Reply:
I will change it like this so we wont invoke that funciton 2 times

Reply:
I tried this but we need to change the logic inside because this `get_import_export_data` is only called when we have force_delete. So there is two possible ways that i can think about it. We can have small meeting and talk about it